packages/core-mobile/app/new/features/browser/components/BrowserInput.tsx

```diff
@@ -0,0 +1,298 @@
+              fontSize: 16,
+              fontFamily: 'Inter-Regular',
+              paddingRight: 24 + HORIZONTAL_MARGIN,
+              backgroundColor: theme.isDark ? '#555557' : theme.colors.$white
+            }}
+          />
+          <Pressable
+            pointerEvents={isFocused ? 'auto' : 'none'}
+            onPress={onClear}
+            style={{
+              position: 'absolute',
+              right: 0,
+              bottom: 0,
+              top: 0,
+              justifyContent: 'center',
+              alignItems: 'center',
+              opacity: urlEntry?.length ? 1 : 0,
+              paddingRight: 12,
+              zIndex: 10
+            }}>
+            <Icons.Action.Clear color={theme.colors.$textSecondary} />
+          </Pressable>
+        </Animated.View>
+
+        <Animated.View
+          style={[
+            contentStyle,
+            {
+              position: 'absolute',
+              top: 0,
+              left: 0,
+              right: 0,
+              bottom: 0,
+              zIndex: 10,
+              borderRadius: 100
+            }
+          ]}>
+          <MaskedProgressBar progress={progress}>
+            <View
+              style={{
+                height: '100%',
+                flexDirection: 'row',
+                alignItems: 'center',
+                justifyContent: 'space-between',
+                backgroundColor: theme.isDark
+                  ? alpha(theme.colors.$white, 0.1)
+                  : alpha(theme.colors.$black, 0.15),
+                borderRadius: 100
+              }}>
+              <Pressable
+                onPress={navigateToTabList}
+                style={{
+                  height: '100%',
+                  paddingHorizontal: 12,
+                  justifyContent: 'center',
+                  alignItems: 'center'
+                }}>
+                <Icons.Navigation.Tabs color={theme.colors.$textPrimary} />
+              </Pressable>
+
+              <Pressable
+                style={{
+                  flex: 1,
+                  height: '100%',
+                  justifyContent: 'center',
+                  alignItems: 'center'
+                }}
+                onPress={() => {
+                  inputRef?.current?.focus()
+                }}>
+                {urlEntry?.length && isValidHttpUrl(urlEntry) ? (
+                  <Text
+                    numberOfLines={1}
+                    style={{
+                      fontFamily: 'Inter-Medium',
+                      fontSize: 14
+                    }}>
+                    <Text
+                      style={{
+                        fontFamily: 'Inter-Medium',
+                        fontSize: 14,
+                        opacity: 0.6
+                      }}>
+                      {parsedUrl?.protocol?.length
+                        ? `${parsedUrl.protocol}://`
+                        : ''}
+                    </Text>
+                    <Text
```

Comment:
nit: extract into render functions

Reply:
thanks for extracting! could you either do useCallback or rename `renderTabListButton` to `tabListButton` since you are doing useMemo?